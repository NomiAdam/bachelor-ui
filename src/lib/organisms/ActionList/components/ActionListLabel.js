import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledLabelWrapper = styled.div`
    width: 100%;
    min-height: 100px;
    display: flex;
    justify-content: center;
    flex-direction: column;
    font-size: 1em;
    cursor: ${({ isAction }) => isAction ? 'pointer' : 'initial' };
`;

const StyledHeading = styled.h4`
    margin: 0;
    padding: 0;
    font-size: 1em;
`;

const StyledParagraph = styled.p`
    font-size: .7em;
`;

const ActionListLabel = ({ header, subHeader, isAction, handleClick }) => (
    <StyledLabelWrapper isAction={isAction} onClick={handleClick}>
        <StyledHeading>{ header }</StyledHeading>
        <StyledParagraph>{ subHeader }</StyledParagraph>
    </StyledLabelWrapper>
);

ActionListLabel.propTypes = {
    /**
     * String definition of label header
     */
    header: PropTypes.string,
    /**
     * String definition of Label subHeader
     */
    subHeader: PropTypes.string,
    /**
     * Boolean indicating whether we use list for actions,
     * Is case of true pointer css will be set and so on
     */
    isAction: PropTypes.bool,
    /**
     * Function to be called upon ListItem click
     */
    handleClick: PropTypes.func,
};

export { StyledLabelWrapper };
export default ActionListLabel;
