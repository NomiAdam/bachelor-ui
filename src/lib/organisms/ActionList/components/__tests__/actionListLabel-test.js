/* eslint-disable no-undef */
import React from 'react';
import ActionListLabel, { StyledLabelWrapper } from '../ActionListLabel';
import {shallow} from 'enzyme';
import 'jest-styled-components';

describe('ActionListLabel', () => {
    it('should be defined', () => {
        expect(ActionListLabel).toBeDefined();
    });
    it('should be match the snapshot', () => {
        expect(shallow(<ActionListLabel/>)).toMatchSnapshot();
    });
    it('should be of type StyledLabelWrapper', () => {
        expect(shallow(<ActionListLabel/>).type()).toBe(StyledLabelWrapper);
    });
    it('should be of type StyledLabelWrapper', () => {
        expect(shallow(<ActionListLabel/>).type()).toBe(StyledLabelWrapper);
    });
    it('should have initial cursor css when isAction is false', () => {
        const wrapper = shallow(<ActionListLabel/>);
        expect(wrapper.find(StyledLabelWrapper)).toHaveStyleRule('cursor','initial');
    });
    it('should have pointer cursor css when isAction is true', () => {
        const wrapper = shallow(<ActionListLabel isAction/>);
        expect(wrapper.find(StyledLabelWrapper)).toHaveStyleRule('cursor','pointer');
    });
});
