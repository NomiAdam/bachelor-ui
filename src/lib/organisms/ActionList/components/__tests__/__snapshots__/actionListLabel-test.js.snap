// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionListLabel should be match the snapshot 1`] = `
<ActionListLabel__StyledLabelWrapper>
  <ActionListLabel__StyledHeading />
  <ActionListLabel__StyledParagraph />
</ActionListLabel__StyledLabelWrapper>
`;
